Assets/Scripts/Classes/CitybuildingManager.cs

```diff
@@ -0,0 +1,314 @@
+        public BuildingUpgrade PlayerFoodGenerator;
+        public BuildingUpgrade PlayerMainCamp;
+        public BuildingUpgrade PlayerMedicSchool;
+        public Resources PlayerResources;
+        public BuildingUpgrade PlayerRobotSchool;
+        public BuildingUpgrade PlayerShooterSchool;
+        public BuildingUpgrade PlayerTitanGenerator;
+        public BuildingUpgrade PlayerWaterGenerator;
+
+        public float PlayerCitySatisfaction => throw new NotImplementedException();
+
+        private BuildingUpgrade[] PlayerBuildings
+        {
+            get
+            {
+                return new[]
+                {
+                    PlayerMainCamp,
+                    PlayerTitanGenerator,
+                    PlayerEnergyGenerator,
+                    PlayerWaterGenerator,
+                    PlayerFoodGenerator,
+                    PlayerFighterSchool,
+                    PlayerShooterSchool,
+                    PlayerDroneSchool,
+                    PlayerMedicSchool,
+                    PlayerRobotSchool
+                };
+            }
+        }
+
+        public void Load() // Called by the GM when loading the game
+        {
+            PlayerResources = new Resources
+            {
+                energy = PlayerPrefs.GetInt("PlayerResources/Energy", 10),
+                titan = PlayerPrefs.GetInt("PlayerResources/Titan", 11),
+                water = PlayerPrefs.GetInt("PlayerResources/Water", 12),
+                food = PlayerPrefs.GetInt("PlayerResources/Food", 13)
+            };
+
+            PlayerMainCamp = MainCamp.upgrades[PlayerPrefs.GetInt("PlayerBuildings/MainCamp", 0)];
+            PlayerTitanGenerator = TitanGenerator.upgrades[PlayerPrefs.GetInt("PlayerBuildings/TitanGenerator", 0)];
+            PlayerEnergyGenerator = EnergyGenerator.upgrades[PlayerPrefs.GetInt("PlayerBuildings/EnergyGenerator", 0)];
+            PlayerWaterGenerator = WaterGenerator.upgrades[PlayerPrefs.GetInt("PlayerBuildings/WaterGenerator", 0)];
+            PlayerFoodGenerator = FoodGenerator.upgrades[PlayerPrefs.GetInt("PlayerBuildings/FoodGenerator", 0)];
+            PlayerFighterSchool = FighterSchool.upgrades[PlayerPrefs.GetInt("PlayerBuildings/FighterSchool", 0)];
+            PlayerShooterSchool = ShooterSchool.upgrades[PlayerPrefs.GetInt("PlayerBuildings/ShooterSchool", 0)];
+            PlayerDroneSchool = DroneSchool.upgrades[PlayerPrefs.GetInt("PlayerBuildings/DroneSchool", 0)];
+            PlayerMedicSchool = MedicSchool.upgrades[PlayerPrefs.GetInt("PlayerBuildings/MedicSchool", 0)];
+            PlayerRobotSchool = RobotSchool.upgrades[PlayerPrefs.GetInt("PlayerBuildings/RobotSchool", 0)];
+        }
+
+        public void Save()
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Update() // Called when skipping to the next day
+        {
+            foreach (var building in PlayerBuildings) PlayerResources += building.output;
+        }
```

Comment:
I'd rename it to something else, because Update() is very misleading at first glance (Might be mistaken with Update() loop from Unity Monobehaviour class)

Reply:
I agree